app/javascript/packs/cms.ts

```diff
@@ -304,18 +279,23 @@ function setUpPjax () {
 }
 
 /**
- * The CMS features 3 dropdown buttons: New button, preview button and save button. Html is rendered
- * by CMS::DropdownHelper and its functionality is given by this method everytime a new template is
+ * The CMS features 4 dropdown buttons: New button, Preview, Publish and Save. Html is rendered
+ * by "/provider/admin/cms/dropdown" partial and its functionality is given by this method everytime a new template is
  * selected (cms-sidebar:update event).
  *
  * The event handler has to be defined with jQuery because at the time this method is called, the
  * CMS content is not yet rendered and document.querySelector would find nothing.
  */
 function setUpDropdownButtonOpen () {
   $(document).on('click', '.dropdown-toggle', (event) => {
```

Comment:
This should be a great opportunity to remove jquery from here. It had to be like that because the dropdown was rendered dynamically and jQuery sets events listener globally.

Now the dropdown renders with the rest of the page and a listener can be set after dom is ready (DOMContentLoaded).

We can also drop `.dropdown-toggle` and use PF's classes. There is a similar snippet already https://github.com/3scale/porta/blob/fix-cms-button-styles/app/javascript/packs/authentication_providers.ts

Reply:
OK, I see the listeners situation. In fact, I didn't fully realize that that Save dropdown was built dynamically :sweat_smile: But yes, after I changed it to dropdown, `buildSaveDropdownButton` seemed useless for sure.

The problem with this dropdown button is that the actual button and the toggle (with the arrow) have the same classes - `pf-c-dropdown__toggle-button`, see https://pf4.patternfly.org/components/dropdown#split-button-primary-action
That's why I had to use a custom class. We could of course search for a ".pf-c-dropdown__toggle-button that has a nested i.fa-caret-down", but I think it's more confusing and complex.

Reply:
Well, I figured out listening for `DOMContentLoaded` is not enough, unfortunately...

Each time another page/layout/etc. is selected in the CMS sidebar, the contents of the page is re-rendered, as well as the new dropdowns (the dropdowns depend on the type of content selected), so we need to attach new event listeners to these new dropdowns.

Reply:
Yes, the open/close logic belongs to the dropdown and it should be imported from the template, not the pack. But OK jQuery is useful in this particular case and it can be defined globally here. The callback can be implemented without jQuery though.

To work around the duplicated class I think an id would be more appropriate as in it identifies a specific button.

Reply:
> The callback can be implemented without jQuery though.

Which callback?... 

> To work around the duplicated class I think an id would be more appropriate as in it identifies a specific button.

But there might be multiple dropdowns on the page, so ID is not good either. Is it that bad to have an additional class?

Reply:
> Which callback?...

This function for the dropdown toggle. It could be done without jQuery:
```js
  const dropdown = event.currentTarget.parentNode
  dropdown.classList.toggle('pf-m-expanded')
```

> But there might be multiple dropdowns on the page, so ID is not good either. Is it that bad to have an additional class?

Fair enough, class wins.

Reply:
> This function for the dropdown toggle. It could be done without jQuery:

I see, thanks!
I was not sure how to go from a jQuery event to a native object. But well, I think as the listener is based on jQuery, it doesn't really matter that much whether the callback uses is or not? I'll keep it as it is for now to avoid another iteration :smiling_face_with_tear: